src/azure-cli-core/azure/cli/core/_profile.py

```diff
@@ -21,6 +21,8 @@
 from azure.cli.core.util import get_file_json, in_cloud_console, open_page_in_browser, can_launch_browser,\
     is_windows, is_wsl
 from azure.cli.core.cloud import get_active_cloud, set_cloud_subscription
+from azure.core.credentials import AccessToken
+from msrestazure.azure_active_directory import MSIAuthentication
```

Comment:
Please delay these imports as they will cause significant latency on client-side commands.

Reply:
How about extracting `MSIAuthenticationWrapper` to `src/azure-cli-core/azure/cli/core/adal_authentication.py` and import it conditionally?

Reply:
done